src/openfermion/transforms/_projection.py

```diff
@@ -0,0 +1,61 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+'''
+Functions to reduce the number of qubits involved
+in modelling a given system
+'''
+from openfermion.ops import QubitOperator
+import numpy
+
+
+def project_onto_sector(operator, qubits, sectors):
+    '''
+    Takes a QubitOperator, and projects out a list
+    of qubits, into either the +1 or -1 sector.
+    Note - this requires knowledge of which sector
+    we wish to project into.
+
+    Input:
+        operator: the QubitOperator to work on
+        qubits: a list of indices of qubits in
+           operator to remove
+        sectors: for each qubit, whether to project
+            into the 0 subspace (<Z>=1) or the
+            1 subspace (<Z>=-1).
+
+    Returns:
+        projected_operator: the resultant operator
+        error: the trace norm of the removed term.
```

Comment:
why is this error returned automatically? It doesn't seem like a crucial thing.

Reply:
Fair point, perhaps we should flag for it? Error provides an upper bound to the error in estimating the eigenvalue of O by the eigenvalue of the projected operator, so I think providing the capability to calculate it is useful, but perhaps returning it always isn't the best idea.

Reply:
Splitting the code into a second function is easy enough, so that works well.

Reply:
Great. I think the doc strings on the first function are out of date now though.

Reply:
Fixed now.